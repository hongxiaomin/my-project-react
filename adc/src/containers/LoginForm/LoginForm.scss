@import url(https://fonts.googleapis.com/css?family=Roboto:300);

.form {
  background: #FFFFFF;
  width: 540px;
  padding: 45px;
  position: relative;
}

.form-input {
  font-family: "Roboto", sans-serif;
  outline: 0;
  width: 60%;
  border: 0;
  // line-height: 25px;
  margin: 0 25% 15px;
  box-sizing: border-box;
  font-size: 14px;
  border-bottom: 1px solid #0087DC;
}

.form .checkbox {
  margin: 0 25% 15px;
}

.form button {
  font-family: "Roboto", sans-serif;
  outline: 0;
  width: 20%;
  border: 0;
  margin: 0 40%;
  color: #FFFFFF;
  background: #0087DC;
  font-size: 14px;
  transition: all 0.3 ease;
  cursor: pointer;
}

.form button:active,
.form button:focus,
.form button:hover {
  background: #0073BB;
}


.submessage {
  font-family: "Roboto", sans-serif;
  font-size: 18px;
  margin: 0 25% 0px;
  color: #0087DC;
}
