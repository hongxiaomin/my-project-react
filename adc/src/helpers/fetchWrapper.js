import { assign } from 'lodash'

export function simpleFetch(path, { method, body, headers } = { method: 'GET', body: {} }) {
  let options = {
    method,
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json',
    },
  }

  if (method !== 'GET') {
    if (headers) {
      options = assign({}, options, {
        headers,
        body,
      })
    } else {
      options = assign({}, options, {
        body: JSON.stringify(body),
      })
    }
  }

  //
  return fetch(`${__HOST__}${path}`, options)
    .then(res => res.json())
    .then((data) => {
      if (data.code === 403002 || data.code === 403001) {
        location.href = '/'
      }

      return data
    })
}

export function jwtFetch({
  token,
  path,
  method = 'GET',
  body = {},
}) {
  if (!token) {
    console.error('Try to call authorized api without token')
    return Promise.resolve({
      tokenIsValid: false,
    })
  }

  let options = {
    method,
    headers: {
      Authorization: token,
      'Content-Type': 'application/json',
    },
  }

  if (method !== 'GET') {
    options = assign({}, options, {
      body: JSON.stringify(body),
    })
  }

  return fetch(`${__URL__}${path}`, options)
}
