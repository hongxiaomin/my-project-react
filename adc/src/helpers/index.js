/* eslint consistent-return: "off"*/
/**
 * [loadState description]
 * @param  {String} attr 'auth' or 'client'
 * @return {String} stringify-object
 */
export const loadState = (name) => {
  try {
    const serializeState = localStorage.getItem(name)
    if (serializeState === null) {
      return undefined
    }
    return JSON.parse(serializeState)
  } catch (err) {
    console.log(err)
    return undefined
  }
}

/**
 * [saveState description]
 * @param  {String} name the attribue e.g.'auth' or 'client'
 * @param  {Object} state the state need to be stored in localStorage
 */
export const saveState = name => (state) => {
  try {
    const serializeState = JSON.stringify(state)
    localStorage.setItem(name, serializeState)
  } catch (err) {
    return console.log(err)
  }
}

/**
 * [setGlobalUrl description]
 * @param {String} [url='172.22.34.220']
 */
export const setGlobalUrl = (url = '172.22.34.220') => {
  global.__URL__ = `http://${url}:50080/api`
  global.__MQTT__ = `wss://${url}:59001`
  global.__HOST__ = `http://${location.hostname}:${location.port}/api/`
  global.__HOST__ = 'http://172.22.34.220:50080/api/'
  // global.__HOST__ = 'http://10.120.138.137:50080/api/'
  // global.__HOST__ = 'http:// 10.147.64.115:5005/'
  // global.__HOST__ = 'http://10.147.76.180:5005/'
  // global.__HOST__ = 'http://172.16.240.42:50080/api/'
  // global.__HOST__ = 'http://10.120.138.138:50080/api/'
}
