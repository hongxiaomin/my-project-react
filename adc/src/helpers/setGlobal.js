const setGlobalUrl = (url) => {
  global.__URL__ = `http://${url}:50080/api`
  global.__MQTT__ = `wss://${url}:59001`
}

export default setGlobalUrl
