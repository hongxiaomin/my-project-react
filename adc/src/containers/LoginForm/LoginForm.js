import React, { Component } from 'react'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { assign } from 'lodash'
import { Button } from 'antd'
import { loadState, saveState } from 'helpers'
import { simpleFetch } from 'helpers/fetchWrapper'
import { setStorage } from 'Utils'
import { authenticate } from 'actions'
import { push } from 'react-router-redux'
import './LoginForm.scss'

const propTypes = {
  goIndex: PropTypes.func.isRequired,
}

export class LoginForm extends Component {
  constructor(props) {
    super(props)
    this.onBtnClick = this.onBtnClick.bind(this)
    // this.togglePasswordKeeping = this.togglePasswordKeeping.bind(this)
    this.handleUsernameChange = this.handleUsernameChange.bind(this)
    this.handlePasswordChange = this.handlePasswordChange.bind(this)
    this.state = {
      rememberMe: true,
      username: '',
      password: '',
      authFailureMessage: '',
      loading: false,
    }
  }

  componentWillMount() {
    const clientState = loadState('client')
    if (clientState) {
      this.setState(assign({}, this.state, {
        username: clientState.username,
      }))
    }

    // 绑定回车键
    window.addEventListener('keypress', this.onKeyDown)
  }

  componentWillUnmount() {
    window.removeEventListener('keypress', this.onKeyDown)
  }

  onKeyDown = (e) => {
    if (e.keyCode === 13) {
      this.onBtnClick()
    }
  }

  onBtnClick() {
    if (!this.state.username || !this.state.password) {
      this.setState({
        authFailureMessage: 'Please enter user name and password.',
      })
      return
    }
    this.setState({
      loading: true,
    })
    simpleFetch('login', {
      method: 'POST',
      body: {
        userId: this.state.username,
        password: this.state.password,
      },
    })
      .then((data) => {
        if (data.code !== 200000) {
          this.setState({
            loading: false,
            authFailureMessage: data.message,
          })
          return
        }
        // 保存信息到sessionstorage
        setStorage('loginInfo', JSON.stringify(data.result))
        // 从result中取出内容
        setStorage('intl', 'zh-CN')
        this.props.goIndex()
      })
  }

  // togglePasswordKeeping() {
  //   this.setState(assign({}, this.state, { rememberMe: !this.state.rememberMe }))
  // }

  handleUsernameChange(e) {
    this.setState(assign({}, this.state, { username: e.target.value }))
  }

  handlePasswordChange(e) {
    this.setState(assign({}, this.state, { password: e.target.value }))
  }

  clearErrorMessage = () => {
    if (this.state.authFailureMessage) {
      this.setState({
        authFailureMessage: '',
      })
    }
  }

  render() {
    const { intl } = this.props

    return (
      <form styleName='form'>
        <span styleName='submessage'><WrapperFormattedMessage id='signIn' /></span>
        <p style={{
          color: 'red',
          textAlign: 'center',
          height: '40px',
        }}
        >{this.state.authFailureMessage}</p>
        <input
          styleName='form-input'
          type='text'
          placeholder={intl.formatMessage({ id: 'userAndLicenseManagement.license.userID' })}
          value={this.state.username}
          onChange={this.handleUsernameChange}
          onFocus={this.clearErrorMessage}
        />
        <br />
        <input
          styleName='form-input'
          type='password'
          placeholder={intl.formatMessage({ id: 'password' })}
          value={this.state.password}
          onChange={this.handlePasswordChange}
          onFocus={this.clearErrorMessage}
        />
        <br />
        {/* <Checkbox
          checked={this.state.rememberMe}
          onChange={this.togglePasswordKeeping}
        >
          记住我
        </Checkbox> */}
        <Button
          type="primary"
          onClick={this.onBtnClick}
          loading={this.state.loading}
        >
          <WrapperFormattedMessage id='signIn' />
        </Button>
      </form>
    )
  }
}

LoginForm.propTypes = propTypes

const mapStateToProps = state => ({
  token: state.auth.token,
  authFailureMessage: state.auth.failureMessage,
})

const mapDispatchToProps = dispatch => ({
  authenticate: (username, password, rememberMe) => {
    // 如果使用者選擇 rememberMe ， 就把目前的 username 存到 localStorage
    if (rememberMe) {
      saveState('client')({
        username,
      })
    } else {
      saveState('client')({
        username: '',
      })
    }
    dispatch(authenticate(username, password))
  },
  goIndex: () => {
    if (JSON.parse(sessionStorage.getItem('loginInfo')).roleIds[0] === 'admin') {
      return dispatch(push('/inline/service'))
    }
    return dispatch(push('/offline/labeling'))
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(LoginForm))
