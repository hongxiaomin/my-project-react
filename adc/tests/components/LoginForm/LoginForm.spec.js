import React from 'react'
import { simpleFetch } from '../../../src/helpers/fetchWrapper'
import { LoginForm } from '../../../src/containers/LoginForm/LoginForm'

describe('(Component) LoginForm', () => {
  const goIndex = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <LoginForm
        goIndex={goIndex}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally', () => {
    const wrapper = mount(
      <LoginForm
        goIndex={goIndex}
      />,
    )
    wrapper.find('input').at(0).simulate('change', { target: { value: 'user' } })
    wrapper.find('input').at(1).simulate('change', { target: { value: 'password' } })
    expect(wrapper.state('username')).toEqual('user')
    expect(wrapper.state('password')).toEqual('password')
    wrapper.find('input').at(0).simulate('click')
    wrapper.find('input').at(1).simulate('click')
    // wrapper.setState({ username: '' })
    wrapper.find('Button').simulate('click')
    // expect(wrapper.state('authFailureMessage')).toEqual('Please enter user name and password.')
    expect(wrapper).toMatchSnapshot()
  })
  it('render password=empty', () => {
    const wrapper = mount(
      <LoginForm
        goIndex={goIndex}
      />,
    )
    wrapper.setState({ password: '', username: '' })
    wrapper.find('Button').simulate('click')
    expect(wrapper.state('authFailureMessage')).toEqual('Please enter user name and password.')
    expect(wrapper).toMatchSnapshot()
  })
})
