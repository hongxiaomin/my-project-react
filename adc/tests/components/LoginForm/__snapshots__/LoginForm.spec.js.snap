// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) LoginForm render normally 1`] = `
<form
  styleName="form"
>
  <span
    styleName="submessage"
  >
    Sign in
  </span>
  <p
    style={
      Object {
        "color": "red",
        "height": "40px",
        "textAlign": "center",
      }
    }
  />
  <input
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="User ID"
    styleName="form-input"
    type="text"
    value=""
  />
  <br />
  <input
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Password"
    styleName="form-input"
    type="password"
    value=""
  />
  <br />
  <Button
    clicked={false}
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    type="primary"
  >
    Sign in
  </Button>
</form>
`;

exports[`(Component) LoginForm render normally 2`] = `
<LoginForm
  goIndex={[Function]}
>
  <form
    styleName="form"
  >
    <span
      styleName="submessage"
    >
      Sign in
    </span>
    <p
      style={
        Object {
          "color": "red",
          "height": "40px",
          "textAlign": "center",
        }
      }
    />
    <input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="User ID"
      styleName="form-input"
      type="text"
      value="user"
    />
    <br />
    <input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      styleName="form-input"
      type="password"
      value="password"
    />
    <br />
    <Button
      clicked={false}
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
      type="primary"
    >
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <span>
          Sign in
        </span>
      </button>
    </Button>
  </form>
</LoginForm>
`;

exports[`(Component) LoginForm render password=empty 1`] = `
<LoginForm
  goIndex={[Function]}
>
  <form
    styleName="form"
  >
    <span
      styleName="submessage"
    >
      Sign in
    </span>
    <p
      style={
        Object {
          "color": "red",
          "height": "40px",
          "textAlign": "center",
        }
      }
    >
      Please enter user name and password.
    </p>
    <input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="User ID"
      styleName="form-input"
      type="text"
      value=""
    />
    <br />
    <input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      styleName="form-input"
      type="password"
      value=""
    />
    <br />
    <Button
      clicked={false}
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
      type="primary"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-clicked"
        onClick={[Function]}
        type="button"
      >
        <span>
          Sign in
        </span>
      </button>
    </Button>
  </form>
</LoginForm>
`;
